src/Phase.hs

```diff
@@ -14,9 +14,6 @@ makePrisms ''Phase
 instance ShortShow Phase where
   shortShow (Phase i) = "p" ++ show i
 
-data PhaseSpec = AllPhases | SpecificPhase !Phase
-makePrisms ''PhaseSpec
-
```

Comment:
`PhaseSpec` no longer exists; a `DeclValidityPtr` now needs to point to a map from `PhaseSpec` to `Scope`, and that's already what a `ScopeSet` is.